Create Transform API returns HTTP 503 when ScriptException thrown

**Elasticsearch version** (`bin/elasticsearch --version`): 7.7.0-SNAPSHOT build 067daeb7

**Plugins installed**: ingest-attachment

**JVM version** (`java -version`): shipped Open JDK

**OS version** (`uname -a` if on a Unix-like system): Windows 10 Pro 64-bit

**Description of the problem including expected versus actual behavior**:

When a transform is created with a `scripted_metric` aggregation on a `text` field, a `ScriptException` occurs, which is expected, but the HTTP status code is 503 Service Unavailable. I would have expected a HTTP status code in the 400-499 range e.g. 400 Bad Request, since the problem is with the request.

**Steps to reproduce**:

 1. Create an index with a text field mapping

```JSON
PUT project
{
  "mappings": {
    "properties": {
      "branches": {
        "type": "text"
      },
      "numberOfCommits": {
        "type": "integer"
      },
      "lastActivity": {
        "type": "date"
      },
    }
  }
}
```

 2. Make a call to the create transform API

```json
PUT /_transform/f-a3d3ab65?pretty=true&error_trace=true 
{
  "description": "f-a3d3ab65",
  "source": {
    "index": ["project"],
    "query": {
      "match_all": {}
    }
  },
  "dest": {
    "index": "transform-f-a3d3ab65"
  },
  "frequency": "1s",
  "pivot": {
    "aggs": {
      "averageCommits": {
        "avg": {
          "field": "numberOfCommits"
        }
      },
      "sumIntoMaster": {
        "scripted_metric": {
          "combine_script": {
            "source": "long sum = 0; for (s in state.masterCommits) { sum += s } return sum"
          },
          "init_script": {
            "source": "state.masterCommits = []"
          },
          "map_script": {
            "source": "state.masterCommits.add(doc.branches.contains('master')? 1 : 0)"
          },
          "reduce_script": {
            "source": "long sum = 0; for (s in states) { sum += s } return sum"
          }
        }
      }
    },
    "group_by": {
      "weekly_activity": {
        "date_histogram": {
          "calendar_interval": "week",
          "field": "lastActivity"
        }
      }
    }
  }
}
```

 3. Observe that the response is a HTTP status code 503 Service Unavailable with the following body

```json
{
  "error" : {
    "root_cause" : [
      {
        "type" : "script_exception",
        "reason" : "runtime error",
        "script_stack" : [
          "org.elasticsearch.index.mapper.TextFieldMapper$TextFieldType.fielddataBuilder(TextFieldMapper.java:761)",
          "org.elasticsearch.index.fielddata.IndexFieldDataService.getForField(IndexFieldDataService.java:116)",
          "org.elasticsearch.index.query.QueryShardContext.lambda$lookup$0(QueryShardContext.java:320)",
          "org.elasticsearch.search.lookup.LeafDocLookup$1.run(LeafDocLookup.java:97)",
          "org.elasticsearch.search.lookup.LeafDocLookup$1.run(LeafDocLookup.java:94)",
          "java.base/java.security.AccessController.doPrivileged(AccessController.java:312)",
          "org.elasticsearch.search.lookup.LeafDocLookup.get(LeafDocLookup.java:94)",
          "org.elasticsearch.search.lookup.LeafDocLookup.get(LeafDocLookup.java:41)",
          "state.masterCommits.add(doc.branches.contains('master')? 1 : 0)",
          "                           ^---- HERE"
        ],
        "script" : "state.masterCommits.add(doc.branches.contains('master')? 1 : 0)",
        "lang" : "painless",
        "position" : {
          "offset" : 27,
          "start" : 0,
          "end" : 63
        },
        "stack_trace" : "ScriptException[runtime error]; nested: IllegalArgumentException[Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [branches] in order to load field data by uninverting the inverted index. Note that this can use significant memory.];\r\n\tat org.elasticsearch.painless.PainlessScript.convertToScriptException(PainlessScript.java:96)\r\n\tat org.elasticsearch.painless.PainlessScript$Script.execute(state.masterCommits.add(doc.branches.contains('master')? 1 : 0):37)\r\n\tat org.elasticsearch.search.aggregations.metrics.ScriptedMetricAggregator$1.collect(ScriptedMetricAggregator.java:82)\r\n\tat org.elasticsearch.search.aggregations.AggregatorFactory$MultiBucketAggregatorWrapper$1.collect(AggregatorFactory.java:142)\r\n\tat org.elasticsearch.search.aggregations.MultiBucketCollector$MultiLeafBucketCollector.collect(MultiBucketCollector.java:200)\r\n\tat org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregator$3.collect(CompositeAggregator.java:406)\r\n\tat org.elasticsearch.search.aggregations.bucket.composite.LongValuesSource$1.collect(LongValuesSource.java:179)\r\n\tat org.elasticsearch.search.aggregations.LeafBucketCollector.collect(LeafBucketCollector.java:82)\r\n\tat org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregator.runDeferredCollections(CompositeAggregator.java:388)\r\n\tat org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregator.buildAggregation(CompositeAggregator.java:142)\r\n\tat org.elasticsearch.search.aggregations.AggregationPhase.execute(AggregationPhase.java:130)\r\n\tat org.elasticsearch.search.query.QueryPhase.execute(QueryPhase.java:157)\r\n\tat org.elasticsearch.indices.IndicesService.lambda$loadIntoContext$21(IndicesService.java:1344)\r\n\tat org.elasticsearch.indices.IndicesService.lambda$cacheShardLevelResult$22(IndicesService.java:1396)\r\n\tat org.elasticsearch.indices.IndicesRequestCache$Loader.load(IndicesRequestCache.java:176)\r\n\tat org.elasticsearch.indices.IndicesRequestCache$Loader.load(IndicesRequestCache.java:159)\r\n\tat org.elasticsearch.common.cache.Cache.computeIfAbsent(Cache.java:433)\r\n\tat org.elasticsearch.indices.IndicesRequestCache.getOrCompute(IndicesRequestCache.java:125)\r\n\tat org.elasticsearch.indices.IndicesService.cacheShardLevelResult(IndicesService.java:1402)\r\n\tat org.elasticsearch.indices.IndicesService.loadIntoContext(IndicesService.java:1341)\r\n\tat org.elasticsearch.search.SearchService.loadOrExecuteQueryPhase(SearchService.java:359)\r\n\tat org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:434)\r\n\tat org.elasticsearch.search.SearchService.access$200(SearchService.java:135)\r\n\tat org.elasticsearch.search.SearchService$2.lambda$onResponse$0(SearchService.java:395)\r\n\tat org.elasticsearch.search.SearchService.lambda$runAsync$0(SearchService.java:411)\r\n\tat org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:44)\r\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:692)\r\n\tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)\r\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)\r\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)\r\n\tat java.base/java.lang.Thread.run(Thread.java:832)\r\nCaused by: java.lang.IllegalArgumentException: Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [branches] in order to load field data by uninverting the inverted index. Note that this can use significant memory.\r\n\tat org.elasticsearch.index.mapper.TextFieldMapper$TextFieldType.fielddataBuilder(TextFieldMapper.java:761)\r\n\tat org.elasticsearch.index.fielddata.IndexFieldDataService.getForField(IndexFieldDataService.java:116)\r\n\tat org.elasticsearch.index.query.QueryShardContext.lambda$lookup$0(QueryShardContext.java:320)\r\n\tat org.elasticsearch.search.lookup.LeafDocLookup$1.run(LeafDocLookup.java:97)\r\n\tat org.elasticsearch.search.lookup.LeafDocLookup$1.run(LeafDocLookup.java:94)\r\n\tat java.base/java.security.AccessController.doPrivileged(AccessController.java:312)\r\n\tat org.elasticsearch.search.lookup.LeafDocLookup.get(LeafDocLookup.java:94)\r\n\tat org.elasticsearch.search.lookup.LeafDocLookup.get(LeafDocLookup.java:41)\r\n\tat org.elasticsearch.painless.PainlessScript$Script.execute(state.masterCommits.add(doc.branches.contains('master')? 1 : 0):28)\r\n\t... 29 more\r\n"
      }
    ],
    "type" : "status_exception",
    "reason" : "Failed to validate configuration",
    "caused_by" : {
      "type" : "status_exception",
      "reason" : "Failed to test query",
      "caused_by" : {
        "type" : "search_phase_execution_exception",
        "reason" : "all shards failed",
        "phase" : "query",
        "grouped" : true,
        "failed_shards" : [
          {
            "shard" : 0,
            "index" : "project",
            "node" : "q9SnhvthS8aXfRfHlsw3Tg",
            "reason" : {
              "type" : "script_exception",
              "reason" : "runtime error",
              "script_stack" : [
                "org.elasticsearch.index.mapper.TextFieldMapper$TextFieldType.fielddataBuilder(TextFieldMapper.java:761)",
                "org.elasticsearch.index.fielddata.IndexFieldDataService.getForField(IndexFieldDataService.java:116)",
                "org.elasticsearch.index.query.QueryShardContext.lambda$lookup$0(QueryShardContext.java:320)",
                "org.elasticsearch.search.lookup.LeafDocLookup$1.run(LeafDocLookup.java:97)",
                "org.elasticsearch.search.lookup.LeafDocLookup$1.run(LeafDocLookup.java:94)",
                "java.base/java.security.AccessController.doPrivileged(AccessController.java:312)",
                "org.elasticsearch.search.lookup.LeafDocLookup.get(LeafDocLookup.java:94)",
                "org.elasticsearch.search.lookup.LeafDocLookup.get(LeafDocLookup.java:41)",
                "state.masterCommits.add(doc.branches.contains('master')? 1 : 0)",
                "                           ^---- HERE"
              ],
              "script" : "state.masterCommits.add(doc.branches.contains('master')? 1 : 0)",
              "lang" : "painless",
              "position" : {
                "offset" : 27,
                "start" : 0,
                "end" : 63
              },
              "caused_by" : {
                "type" : "illegal_argument_exception",
                "reason" : "Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [branches] in order to load field data by uninverting the inverted index. Note that this can use significant memory.",
                "stack_trace" : "java.lang.IllegalArgumentException: Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [branches] in order to load field data by uninverting the inverted index. Note that this can use significant memory.\r\n\tat org.elasticsearch.index.mapper.TextFieldMapper$TextFieldType.fielddataBuilder(TextFieldMapper.java:761)\r\n\tat org.elasticsearch.index.fielddata.IndexFieldDataService.getForField(IndexFieldDataService.java:116)\r\n\tat org.elasticsearch.index.query.QueryShardContext.lambda$lookup$0(QueryShardContext.java:320)\r\n\tat org.elasticsearch.search.lookup.LeafDocLookup$1.run(LeafDocLookup.java:97)\r\n\tat org.elasticsearch.search.lookup.LeafDocLookup$1.run(LeafDocLookup.java:94)\r\n\tat java.base/java.security.AccessController.doPrivileged(AccessController.java:312)\r\n\tat org.elasticsearch.search.lookup.LeafDocLookup.get(LeafDocLookup.java:94)\r\n\tat org.elasticsearch.search.lookup.LeafDocLookup.get(LeafDocLookup.java:41)\r\n\tat org.elasticsearch.painless.PainlessScript$Script.execute(state.masterCommits.add(doc.branches.contains('master')? 1 : 0):28)\r\n\tat org.elasticsearch.search.aggregations.metrics.ScriptedMetricAggregator$1.collect(ScriptedMetricAggregator.java:82)\r\n\tat org.elasticsearch.search.aggregations.AggregatorFactory$MultiBucketAggregatorWrapper$1.collect(AggregatorFactory.java:142)\r\n\tat org.elasticsearch.search.aggregations.MultiBucketCollector$MultiLeafBucketCollector.collect(MultiBucketCollector.java:200)\r\n\tat org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregator$3.collect(CompositeAggregator.java:406)\r\n\tat org.elasticsearch.search.aggregations.bucket.composite.LongValuesSource$1.collect(LongValuesSource.java:179)\r\n\tat org.elasticsearch.search.aggregations.LeafBucketCollector.collect(LeafBucketCollector.java:82)\r\n\tat org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregator.runDeferredCollections(CompositeAggregator.java:388)\r\n\tat org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregator.buildAggregation(CompositeAggregator.java:142)\r\n\tat org.elasticsearch.search.aggregations.AggregationPhase.execute(AggregationPhase.java:130)\r\n\tat org.elasticsearch.search.query.QueryPhase.execute(QueryPhase.java:157)\r\n\tat org.elasticsearch.indices.IndicesService.lambda$loadIntoContext$21(IndicesService.java:1344)\r\n\tat org.elasticsearch.indices.IndicesService.lambda$cacheShardLevelResult$22(IndicesService.java:1396)\r\n\tat org.elasticsearch.indices.IndicesRequestCache$Loader.load(IndicesRequestCache.java:176)\r\n\tat org.elasticsearch.indices.IndicesRequestCache$Loader.load(IndicesRequestCache.java:159)\r\n\tat org.elasticsearch.common.cache.Cache.computeIfAbsent(Cache.java:433)\r\n\tat org.elasticsearch.indices.IndicesRequestCache.getOrCompute(IndicesRequestCache.java:125)\r\n\tat org.elasticsearch.indices.IndicesService.cacheShardLevelResult(IndicesService.java:1402)\r\n\tat org.elasticsearch.indices.IndicesService.loadIntoContext(IndicesService.java:1341)\r\n\tat org.elasticsearch.search.SearchService.loadOrExecuteQueryPhase(SearchService.java:359)\r\n\tat org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:434)\r\n\tat org.elasticsearch.search.SearchService.access$200(SearchService.java:135)\r\n\tat org.elasticsearch.search.SearchService$2.lambda$onResponse$0(SearchService.java:395)\r\n\tat org.elasticsearch.search.SearchService.lambda$runAsync$0(SearchService.java:411)\r\n\tat org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:44)\r\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:692)\r\n\tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)\r\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)\r\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)\r\n\tat java.base/java.lang.Thread.run(Thread.java:832)\r\n"
              },
              "stack_trace" : "ScriptException[runtime error]; nested: IllegalArgumentException[Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [branches] in order to load field data by uninverting the inverted index. Note that this can use significant memory.];\r\n\tat org.elasticsearch.painless.PainlessScript.convertToScriptException(PainlessScript.java:96)\r\n\tat org.elasticsearch.painless.PainlessScript$Script.execute(state.masterCommits.add(doc.branches.contains('master')? 1 : 0):37)\r\n\tat org.elasticsearch.search.aggregations.metrics.ScriptedMetricAggregator$1.collect(ScriptedMetricAggregator.java:82)\r\n\tat org.elasticsearch.search.aggregations.AggregatorFactory$MultiBucketAggregatorWrapper$1.collect(AggregatorFactory.java:142)\r\n\tat org.elasticsearch.search.aggregations.MultiBucketCollector$MultiLeafBucketCollector.collect(MultiBucketCollector.java:200)\r\n\tat org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregator$3.collect(CompositeAggregator.java:406)\r\n\tat org.elasticsearch.search.aggregations.bucket.composite.LongValuesSource$1.collect(LongValuesSource.java:179)\r\n\tat org.elasticsearch.search.aggregations.LeafBucketCollector.collect(LeafBucketCollector.java:82)\r\n\tat org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregator.runDeferredCollections(CompositeAggregator.java:388)\r\n\tat org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregator.buildAggregation(CompositeAggregator.java:142)\r\n\tat org.elasticsearch.search.aggregations.AggregationPhase.execute(AggregationPhase.java:130)\r\n\tat org.elasticsearch.search.query.QueryPhase.execute(QueryPhase.java:157)\r\n\tat org.elasticsearch.indices.IndicesService.lambda$loadIntoContext$21(IndicesService.java:1344)\r\n\tat org.elasticsearch.indices.IndicesService.lambda$cacheShardLevelResult$22(IndicesService.java:1396)\r\n\tat org.elasticsearch.indices.IndicesRequestCache$Loader.load(IndicesRequestCache.java:176)\r\n\tat org.elasticsearch.indices.IndicesRequestCache$Loader.load(IndicesRequestCache.java:159)\r\n\tat org.elasticsearch.common.cache.Cache.computeIfAbsent(Cache.java:433)\r\n\tat org.elasticsearch.indices.IndicesRequestCache.getOrCompute(IndicesRequestCache.java:125)\r\n\tat org.elasticsearch.indices.IndicesService.cacheShardLevelResult(IndicesService.java:1402)\r\n\tat org.elasticsearch.indices.IndicesService.loadIntoContext(IndicesService.java:1341)\r\n\tat org.elasticsearch.search.SearchService.loadOrExecuteQueryPhase(SearchService.java:359)\r\n\tat org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:434)\r\n\tat org.elasticsearch.search.SearchService.access$200(SearchService.java:135)\r\n\tat org.elasticsearch.search.SearchService$2.lambda$onResponse$0(SearchService.java:395)\r\n\tat org.elasticsearch.search.SearchService.lambda$runAsync$0(SearchService.java:411)\r\n\tat org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:44)\r\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:692)\r\n\tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)\r\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)\r\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)\r\n\tat java.base/java.lang.Thread.run(Thread.java:832)\r\nCaused by: java.lang.IllegalArgumentException: Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [branches] in order to load field data by uninverting the inverted index. Note that this can use significant memory.\r\n\tat org.elasticsearch.index.mapper.TextFieldMapper$TextFieldType.fielddataBuilder(TextFieldMapper.java:761)\r\n\tat org.elasticsearch.index.fielddata.IndexFieldDataService.getForField(IndexFieldDataService.java:116)\r\n\tat org.elasticsearch.index.query.QueryShardContext.lambda$lookup$0(QueryShardContext.java:320)\r\n\tat org.elasticsearch.search.lookup.LeafDocLookup$1.run(LeafDocLookup.java:97)\r\n\tat org.elasticsearch.search.lookup.LeafDocLookup$1.run(LeafDocLookup.java:94)\r\n\tat java.base/java.security.AccessController.doPrivileged(AccessController.java:312)\r\n\tat org.elasticsearch.search.lookup.LeafDocLookup.get(LeafDocLookup.java:94)\r\n\tat org.elasticsearch.search.lookup.LeafDocLookup.get(LeafDocLookup.java:41)\r\n\tat org.elasticsearch.painless.PainlessScript$Script.execute(state.masterCommits.add(doc.branches.contains('master')? 1 : 0):28)\r\n\t... 29 more\r\n"
            }
          }
        ],
        "stack_trace" : "Failed to execute phase [query], all shards failed; shardFailures {[q9SnhvthS8aXfRfHlsw3Tg][project][0]: RemoteTransportException[[xpack-node-722dce9200][127.0.0.1:9300][indices:data/read/search[phase/query]]]; nested: ScriptException[runtime error]; nested: IllegalArgumentException[Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [branches] in order to load field data by uninverting the inverted index. Note that this can use significant memory.]; }{[q9SnhvthS8aXfRfHlsw3Tg][project][1]: RemoteTransportException[[xpack-node-722dce9200][127.0.0.1:9300][indices:data/read/search[phase/query]]]; nested: ScriptException[runtime error]; nested: IllegalArgumentException[Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [branches] in order to load field data by uninverting the inverted index. Note that this can use significant memory.]; }\r\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.onPhaseFailure(AbstractSearchAsyncAction.java:551)\r\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.executeNextPhase(AbstractSearchAsyncAction.java:309)\r\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.onPhaseDone(AbstractSearchAsyncAction.java:580)\r\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.onShardFailure(AbstractSearchAsyncAction.java:393)\r\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.access$100(AbstractSearchAsyncAction.java:68)\r\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction$1.onFailure(AbstractSearchAsyncAction.java:245)\r\n\tat org.elasticsearch.action.search.SearchExecutionStatsCollector.onFailure(SearchExecutionStatsCollector.java:73)\r\n\tat org.elasticsearch.action.ActionListenerResponseHandler.handleException(ActionListenerResponseHandler.java:59)\r\n\tat org.elasticsearch.action.search.SearchTransportService$ConnectionCountingHandler.handleException(SearchTransportService.java:402)\r\n\tat org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleException(TransportService.java:1139)\r\n\tat org.elasticsearch.transport.TransportService$DirectResponseChannel.processException(TransportService.java:1248)\r\n\tat org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1222)\r\n\tat org.elasticsearch.transport.TaskTransportChannel.sendResponse(TaskTransportChannel.java:60)\r\n\tat org.elasticsearch.action.support.ChannelActionListener.onFailure(ChannelActionListener.java:56)\r\n\tat org.elasticsearch.search.SearchService.lambda$runAsync$0(SearchService.java:413)\r\n\tat org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:44)\r\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:692)\r\n\tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)\r\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)\r\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)\r\n\tat java.base/java.lang.Thread.run(Thread.java:832)\r\nCaused by: ScriptException[runtime error]; nested: IllegalArgumentException[Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [branches] in order to load field data by uninverting the inverted index. Note that this can use significant memory.];\r\n\tat org.elasticsearch.painless.PainlessScript.convertToScriptException(PainlessScript.java:96)\r\n\tat org.elasticsearch.painless.PainlessScript$Script.execute(state.masterCommits.add(doc.branches.contains('master')? 1 : 0):37)\r\n\tat org.elasticsearch.search.aggregations.metrics.ScriptedMetricAggregator$1.collect(ScriptedMetricAggregator.java:82)\r\n\tat org.elasticsearch.search.aggregations.AggregatorFactory$MultiBucketAggregatorWrapper$1.collect(AggregatorFactory.java:142)\r\n\tat org.elasticsearch.search.aggregations.MultiBucketCollector$MultiLeafBucketCollector.collect(MultiBucketCollector.java:200)\r\n\tat org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregator$3.collect(CompositeAggregator.java:406)\r\n\tat org.elasticsearch.search.aggregations.bucket.composite.LongValuesSource$1.collect(LongValuesSource.java:179)\r\n\tat org.elasticsearch.search.aggregations.LeafBucketCollector.collect(LeafBucketCollector.java:82)\r\n\tat org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregator.runDeferredCollections(CompositeAggregator.java:388)\r\n\tat org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregator.buildAggregation(CompositeAggregator.java:142)\r\n\tat org.elasticsearch.search.aggregations.AggregationPhase.execute(AggregationPhase.java:130)\r\n\tat org.elasticsearch.search.query.QueryPhase.execute(QueryPhase.java:157)\r\n\tat org.elasticsearch.indices.IndicesService.lambda$loadIntoContext$21(IndicesService.java:1344)\r\n\tat org.elasticsearch.indices.IndicesService.lambda$cacheShardLevelResult$22(IndicesService.java:1396)\r\n\tat org.elasticsearch.indices.IndicesRequestCache$Loader.load(IndicesRequestCache.java:176)\r\n\tat org.elasticsearch.indices.IndicesRequestCache$Loader.load(IndicesRequestCache.java:159)\r\n\tat org.elasticsearch.common.cache.Cache.computeIfAbsent(Cache.java:433)\r\n\tat org.elasticsearch.indices.IndicesRequestCache.getOrCompute(IndicesRequestCache.java:125)\r\n\tat org.elasticsearch.indices.IndicesService.cacheShardLevelResult(IndicesService.java:1402)\r\n\tat org.elasticsearch.indices.IndicesService.loadIntoContext(IndicesService.java:1341)\r\n\tat org.elasticsearch.search.SearchService.loadOrExecuteQueryPhase(SearchService.java:359)\r\n\tat org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:434)\r\n\tat org.elasticsearch.search.SearchService.access$200(SearchService.java:135)\r\n\tat org.elasticsearch.search.SearchService$2.lambda$onResponse$0(SearchService.java:395)\r\n\tat org.elasticsearch.search.SearchService.lambda$runAsync$0(SearchService.java:411)\r\n\t... 6 more\r\nCaused by: java.lang.IllegalArgumentException: Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [branches] in order to load field data by uninverting the inverted index. Note that this can use significant memory.\r\n\tat org.elasticsearch.index.mapper.TextFieldMapper$TextFieldType.fielddataBuilder(TextFieldMapper.java:761)\r\n\tat org.elasticsearch.index.fielddata.IndexFieldDataService.getForField(IndexFieldDataService.java:116)\r\n\tat org.elasticsearch.index.query.QueryShardContext.lambda$lookup$0(QueryShardContext.java:320)\r\n\tat org.elasticsearch.search.lookup.LeafDocLookup$1.run(LeafDocLookup.java:97)\r\n\tat org.elasticsearch.search.lookup.LeafDocLookup$1.run(LeafDocLookup.java:94)\r\n\tat java.base/java.security.AccessController.doPrivileged(AccessController.java:312)\r\n\tat org.elasticsearch.search.lookup.LeafDocLookup.get(LeafDocLookup.java:94)\r\n\tat org.elasticsearch.search.lookup.LeafDocLookup.get(LeafDocLookup.java:41)\r\n\tat org.elasticsearch.painless.PainlessScript$Script.execute(state.masterCommits.add(doc.branches.contains('master')? 1 : 0):28)\r\n\t... 29 more\r\n"
      },
      "stack_trace" : "ElasticsearchStatusException[Failed to test query]; nested: SearchPhaseExecutionException[all shards failed]; nested: ScriptException[runtime error]; nested: IllegalArgumentException[Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [branches] in order to load field data by uninverting the inverted index. Note that this can use significant memory.];\r\n\tat org.elasticsearch.xpack.transform.transforms.pivot.Pivot.lambda$validateQuery$1(Pivot.java:102)\r\n\tat org.elasticsearch.action.ActionListener$1.onFailure(ActionListener.java:71)\r\n\tat org.elasticsearch.action.support.TransportAction$1.onFailure(TransportAction.java:79)\r\n\tat org.elasticsearch.action.support.ContextPreservingActionListener.onFailure(ContextPreservingActionListener.java:50)\r\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.raisePhaseFailure(AbstractSearchAsyncAction.java:571)\r\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.onPhaseFailure(AbstractSearchAsyncAction.java:551)\r\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.executeNextPhase(AbstractSearchAsyncAction.java:309)\r\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.onPhaseDone(AbstractSearchAsyncAction.java:580)\r\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.onShardFailure(AbstractSearchAsyncAction.java:393)\r\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.access$100(AbstractSearchAsyncAction.java:68)\r\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction$1.onFailure(AbstractSearchAsyncAction.java:245)\r\n\tat org.elasticsearch.action.search.SearchExecutionStatsCollector.onFailure(SearchExecutionStatsCollector.java:73)\r\n\tat org.elasticsearch.action.ActionListenerResponseHandler.handleException(ActionListenerResponseHandler.java:59)\r\n\tat org.elasticsearch.action.search.SearchTransportService$ConnectionCountingHandler.handleException(SearchTransportService.java:402)\r\n\tat org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleException(TransportService.java:1139)\r\n\tat org.elasticsearch.transport.TransportService$DirectResponseChannel.processException(TransportService.java:1248)\r\n\tat org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1222)\r\n\tat org.elasticsearch.transport.TaskTransportChannel.sendResponse(TaskTransportChannel.java:60)\r\n\tat org.elasticsearch.action.support.ChannelActionListener.onFailure(ChannelActionListener.java:56)\r\n\tat org.elasticsearch.search.SearchService.lambda$runAsync$0(SearchService.java:413)\r\n\tat org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:44)\r\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:692)\r\n\tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)\r\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)\r\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)\r\n\tat java.base/java.lang.Thread.run(Thread.java:832)\r\nCaused by: Failed to execute phase [query], all shards failed; shardFailures {[q9SnhvthS8aXfRfHlsw3Tg][project][0]: RemoteTransportException[[xpack-node-722dce9200][127.0.0.1:9300][indices:data/read/search[phase/query]]]; nested: ScriptException[runtime error]; nested: IllegalArgumentException[Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [branches] in order to load field data by uninverting the inverted index. Note that this can use significant memory.]; }{[q9SnhvthS8aXfRfHlsw3Tg][project][1]: RemoteTransportException[[xpack-node-722dce9200][127.0.0.1:9300][indices:data/read/search[phase/query]]]; nested: ScriptException[runtime error]; nested: IllegalArgumentException[Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [branches] in order to load field data by uninverting the inverted index. Note that this can use significant memory.]; }\r\n\t... 21 more\r\nCaused by: ScriptException[runtime error]; nested: IllegalArgumentException[Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [branches] in order to load field data by uninverting the inverted index. Note that this can use significant memory.];\r\n\tat org.elasticsearch.painless.PainlessScript.convertToScriptException(PainlessScript.java:96)\r\n\tat org.elasticsearch.painless.PainlessScript$Script.execute(state.masterCommits.add(doc.branches.contains('master')? 1 : 0):37)\r\n\tat org.elasticsearch.search.aggregations.metrics.ScriptedMetricAggregator$1.collect(ScriptedMetricAggregator.java:82)\r\n\tat org.elasticsearch.search.aggregations.AggregatorFactory$MultiBucketAggregatorWrapper$1.collect(AggregatorFactory.java:142)\r\n\tat org.elasticsearch.search.aggregations.MultiBucketCollector$MultiLeafBucketCollector.collect(MultiBucketCollector.java:200)\r\n\tat org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregator$3.collect(CompositeAggregator.java:406)\r\n\tat org.elasticsearch.search.aggregations.bucket.composite.LongValuesSource$1.collect(LongValuesSource.java:179)\r\n\tat org.elasticsearch.search.aggregations.LeafBucketCollector.collect(LeafBucketCollector.java:82)\r\n\tat org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregator.runDeferredCollections(CompositeAggregator.java:388)\r\n\tat org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregator.buildAggregation(CompositeAggregator.java:142)\r\n\tat org.elasticsearch.search.aggregations.AggregationPhase.execute(AggregationPhase.java:130)\r\n\tat org.elasticsearch.search.query.QueryPhase.execute(QueryPhase.java:157)\r\n\tat org.elasticsearch.indices.IndicesService.lambda$loadIntoContext$21(IndicesService.java:1344)\r\n\tat org.elasticsearch.indices.IndicesService.lambda$cacheShardLevelResult$22(IndicesService.java:1396)\r\n\tat org.elasticsearch.indices.IndicesRequestCache$Loader.load(IndicesRequestCache.java:176)\r\n\tat org.elasticsearch.indices.IndicesRequestCache$Loader.load(IndicesRequestCache.java:159)\r\n\tat org.elasticsearch.common.cache.Cache.computeIfAbsent(Cache.java:433)\r\n\tat org.elasticsearch.indices.IndicesRequestCache.getOrCompute(IndicesRequestCache.java:125)\r\n\tat org.elasticsearch.indices.IndicesService.cacheShardLevelResult(IndicesService.java:1402)\r\n\tat org.elasticsearch.indices.IndicesService.loadIntoContext(IndicesService.java:1341)\r\n\tat org.elasticsearch.search.SearchService.loadOrExecuteQueryPhase(SearchService.java:359)\r\n\tat org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:434)\r\n\tat org.elasticsearch.search.SearchService.access$200(SearchService.java:135)\r\n\tat org.elasticsearch.search.SearchService$2.lambda$onResponse$0(SearchService.java:395)\r\n\tat org.elasticsearch.search.SearchService.lambda$runAsync$0(SearchService.java:411)\r\n\t... 6 more\r\nCaused by: java.lang.IllegalArgumentException: Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [branches] in order to load field data by uninverting the inverted index. Note that this can use significant memory.\r\n\tat org.elasticsearch.index.mapper.TextFieldMapper$TextFieldType.fielddataBuilder(TextFieldMapper.java:761)\r\n\tat org.elasticsearch.index.fielddata.IndexFieldDataService.getForField(IndexFieldDataService.java:116)\r\n\tat org.elasticsearch.index.query.QueryShardContext.lambda$lookup$0(QueryShardContext.java:320)\r\n\tat org.elasticsearch.search.lookup.LeafDocLookup$1.run(LeafDocLookup.java:97)\r\n\tat org.elasticsearch.search.lookup.LeafDocLookup$1.run(LeafDocLookup.java:94)\r\n\tat java.base/java.security.AccessController.doPrivileged(AccessController.java:312)\r\n\tat org.elasticsearch.search.lookup.LeafDocLookup.get(LeafDocLookup.java:94)\r\n\tat org.elasticsearch.search.lookup.LeafDocLookup.get(LeafDocLookup.java:41)\r\n\tat org.elasticsearch.painless.PainlessScript$Script.execute(state.masterCommits.add(doc.branches.contains('master')? 1 : 0):28)\r\n\t... 29 more\r\n"
    },
    "stack_trace" : "ElasticsearchStatusException[Failed to validate configuration]; nested: ElasticsearchStatusException[Failed to test query]; nested: SearchPhaseExecutionException[all shards failed]; nested: ScriptException[runtime error]; nested: IllegalArgumentException[Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [branches] in order to load field data by uninverting the inverted index. Note that this can use significant memory.];\r\n\tat org.elasticsearch.xpack.transform.action.TransportPutTransformAction.lambda$putTransform$5(TransportPutTransformAction.java:308)\r\n\tat org.elasticsearch.action.ActionListener$1.onFailure(ActionListener.java:71)\r\n\tat org.elasticsearch.xpack.transform.transforms.pivot.Pivot.lambda$validateQuery$1(Pivot.java:102)\r\n\tat org.elasticsearch.action.ActionListener$1.onFailure(ActionListener.java:71)\r\n\tat org.elasticsearch.action.support.TransportAction$1.onFailure(TransportAction.java:79)\r\n\tat org.elasticsearch.action.support.ContextPreservingActionListener.onFailure(ContextPreservingActionListener.java:50)\r\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.raisePhaseFailure(AbstractSearchAsyncAction.java:571)\r\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.onPhaseFailure(AbstractSearchAsyncAction.java:551)\r\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.executeNextPhase(AbstractSearchAsyncAction.java:309)\r\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.onPhaseDone(AbstractSearchAsyncAction.java:580)\r\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.onShardFailure(AbstractSearchAsyncAction.java:393)\r\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.access$100(AbstractSearchAsyncAction.java:68)\r\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction$1.onFailure(AbstractSearchAsyncAction.java:245)\r\n\tat org.elasticsearch.action.search.SearchExecutionStatsCollector.onFailure(SearchExecutionStatsCollector.java:73)\r\n\tat org.elasticsearch.action.ActionListenerResponseHandler.handleException(ActionListenerResponseHandler.java:59)\r\n\tat org.elasticsearch.action.search.SearchTransportService$ConnectionCountingHandler.handleException(SearchTransportService.java:402)\r\n\tat org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleException(TransportService.java:1139)\r\n\tat org.elasticsearch.transport.TransportService$DirectResponseChannel.processException(TransportService.java:1248)\r\n\tat org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1222)\r\n\tat org.elasticsearch.transport.TaskTransportChannel.sendResponse(TaskTransportChannel.java:60)\r\n\tat org.elasticsearch.action.support.ChannelActionListener.onFailure(ChannelActionListener.java:56)\r\n\tat org.elasticsearch.search.SearchService.lambda$runAsync$0(SearchService.java:413)\r\n\tat org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:44)\r\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:692)\r\n\tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)\r\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)\r\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)\r\n\tat java.base/java.lang.Thread.run(Thread.java:832)\r\nCaused by: ElasticsearchStatusException[Failed to test query]; nested: SearchPhaseExecutionException[all shards failed]; nested: ScriptException[runtime error]; nested: IllegalArgumentException[Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [branches] in order to load field data by uninverting the inverted index. Note that this can use significant memory.];\r\n\t... 26 more\r\nCaused by: Failed to execute phase [query], all shards failed; shardFailures {[q9SnhvthS8aXfRfHlsw3Tg][project][0]: RemoteTransportException[[xpack-node-722dce9200][127.0.0.1:9300][indices:data/read/search[phase/query]]]; nested: ScriptException[runtime error]; nested: IllegalArgumentException[Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [branches] in order to load field data by uninverting the inverted index. Note that this can use significant memory.]; }{[q9SnhvthS8aXfRfHlsw3Tg][project][1]: RemoteTransportException[[xpack-node-722dce9200][127.0.0.1:9300][indices:data/read/search[phase/query]]]; nested: ScriptException[runtime error]; nested: IllegalArgumentException[Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [branches] in order to load field data by uninverting the inverted index. Note that this can use significant memory.]; }\r\n\t... 21 more\r\nCaused by: ScriptException[runtime error]; nested: IllegalArgumentException[Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [branches] in order to load field data by uninverting the inverted index. Note that this can use significant memory.];\r\n\tat org.elasticsearch.painless.PainlessScript.convertToScriptException(PainlessScript.java:96)\r\n\tat org.elasticsearch.painless.PainlessScript$Script.execute(state.masterCommits.add(doc.branches.contains('master')? 1 : 0):37)\r\n\tat org.elasticsearch.search.aggregations.metrics.ScriptedMetricAggregator$1.collect(ScriptedMetricAggregator.java:82)\r\n\tat org.elasticsearch.search.aggregations.AggregatorFactory$MultiBucketAggregatorWrapper$1.collect(AggregatorFactory.java:142)\r\n\tat org.elasticsearch.search.aggregations.MultiBucketCollector$MultiLeafBucketCollector.collect(MultiBucketCollector.java:200)\r\n\tat org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregator$3.collect(CompositeAggregator.java:406)\r\n\tat org.elasticsearch.search.aggregations.bucket.composite.LongValuesSource$1.collect(LongValuesSource.java:179)\r\n\tat org.elasticsearch.search.aggregations.LeafBucketCollector.collect(LeafBucketCollector.java:82)\r\n\tat org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregator.runDeferredCollections(CompositeAggregator.java:388)\r\n\tat org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregator.buildAggregation(CompositeAggregator.java:142)\r\n\tat org.elasticsearch.search.aggregations.AggregationPhase.execute(AggregationPhase.java:130)\r\n\tat org.elasticsearch.search.query.QueryPhase.execute(QueryPhase.java:157)\r\n\tat org.elasticsearch.indices.IndicesService.lambda$loadIntoContext$21(IndicesService.java:1344)\r\n\tat org.elasticsearch.indices.IndicesService.lambda$cacheShardLevelResult$22(IndicesService.java:1396)\r\n\tat org.elasticsearch.indices.IndicesRequestCache$Loader.load(IndicesRequestCache.java:176)\r\n\tat org.elasticsearch.indices.IndicesRequestCache$Loader.load(IndicesRequestCache.java:159)\r\n\tat org.elasticsearch.common.cache.Cache.computeIfAbsent(Cache.java:433)\r\n\tat org.elasticsearch.indices.IndicesRequestCache.getOrCompute(IndicesRequestCache.java:125)\r\n\tat org.elasticsearch.indices.IndicesService.cacheShardLevelResult(IndicesService.java:1402)\r\n\tat org.elasticsearch.indices.IndicesService.loadIntoContext(IndicesService.java:1341)\r\n\tat org.elasticsearch.search.SearchService.loadOrExecuteQueryPhase(SearchService.java:359)\r\n\tat org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:434)\r\n\tat org.elasticsearch.search.SearchService.access$200(SearchService.java:135)\r\n\tat org.elasticsearch.search.SearchService$2.lambda$onResponse$0(SearchService.java:395)\r\n\tat org.elasticsearch.search.SearchService.lambda$runAsync$0(SearchService.java:411)\r\n\t... 6 more\r\nCaused by: java.lang.IllegalArgumentException: Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [branches] in order to load field data by uninverting the inverted index. Note that this can use significant memory.\r\n\tat org.elasticsearch.index.mapper.TextFieldMapper$TextFieldType.fielddataBuilder(TextFieldMapper.java:761)\r\n\tat org.elasticsearch.index.fielddata.IndexFieldDataService.getForField(IndexFieldDataService.java:116)\r\n\tat org.elasticsearch.index.query.QueryShardContext.lambda$lookup$0(QueryShardContext.java:320)\r\n\tat org.elasticsearch.search.lookup.LeafDocLookup$1.run(LeafDocLookup.java:97)\r\n\tat org.elasticsearch.search.lookup.LeafDocLookup$1.run(LeafDocLookup.java:94)\r\n\tat java.base/java.security.AccessController.doPrivileged(AccessController.java:312)\r\n\tat org.elasticsearch.search.lookup.LeafDocLookup.get(LeafDocLookup.java:94)\r\n\tat org.elasticsearch.search.lookup.LeafDocLookup.get(LeafDocLookup.java:41)\r\n\tat org.elasticsearch.painless.PainlessScript$Script.execute(state.masterCommits.add(doc.branches.contains('master')? 1 : 0):28)\r\n\t... 29 more\r\n"
  },
  "status" : 503
}
```
  Would expect a response in the 400-499 range. With a 503 response, the official language clients will treat this as an error that can be retried.

--- a/x-pack/plugin/src/test/resources/rest-api-spec/test/transform/transforms_crud.yml
+++ b/x-pack/plugin/src/test/resources/rest-api-spec/test/transform/transforms_crud.yml
@@ -637,3 +637,36 @@ setup:
             "description": "yaml test transform on airline-data"
           }
   - match: {acknowledged: true}
+---
+"Test put transform with bad query due to script compilation":
+  - do:
+      catch: bad_request
+      transform.put_transform:
+        transform_id: "airline-transform-with-bad-scripts"
+        body: >
+          {
+            "source": { "index": "airline-data" },
+            "dest": { "index": "airline-data-by-airline-with-bad-scripts" },
+            "pivot": {
+              "group_by": { "airline": {"terms": {"field": "airline"}}},
+              "aggs": {
+                "sumState": {
+                  "scripted_metric": {
+                    "combine_script": {
+                      "source": "foobar boom"
+                    },
+                    "init_script": {
+                      "source": "state.state = []"
+                    },
+                    "map_script": {
+                      "source": "state.state.add(1)"
+                    },
+                    "reduce_script": {
+                      "source": "long sum = 0; for (s in states) { sum += s } return sum"
+                    }
+                  }
+                }
+              }
+            },
+            "description": "yaml test transform on airline-data"
+          }

--- a/x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/pivot/Pivot.java
+++ b/x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/pivot/Pivot.java
@@ -9,6 +9,7 @@ package org.elasticsearch.xpack.transform.transforms.pivot;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.apache.logging.log4j.message.ParameterizedMessage;
+import org.elasticsearch.ElasticsearchException;
 import org.elasticsearch.ElasticsearchStatusException;
 import org.elasticsearch.action.ActionListener;
 import org.elasticsearch.action.search.SearchAction;
@@ -28,6 +29,7 @@ import org.elasticsearch.search.aggregations.PipelineAggregationBuilder;
 import org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregation;
 import org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregationBuilder;
 import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.elasticsearch.xpack.core.ml.utils.ExceptionsHelper;
 import org.elasticsearch.xpack.core.transform.TransformMessages;
 import org.elasticsearch.xpack.core.transform.transforms.SourceConfig;
 import org.elasticsearch.xpack.core.transform.transforms.TransformIndexerStats;
@@ -99,7 +101,15 @@ public class Pivot {
                 return;
             }
             listener.onResponse(true);
-        }, e -> listener.onFailure(new ElasticsearchStatusException("Failed to test query", RestStatus.SERVICE_UNAVAILABLE, e))));
+        }, e -> {
+            Throwable unwrapped = ExceptionsHelper.unwrapCause(e);
+            RestStatus status = unwrapped instanceof ElasticsearchException ?
+                ((ElasticsearchException)unwrapped).status() :
+                RestStatus.SERVICE_UNAVAILABLE;
+            listener.onFailure(new ElasticsearchStatusException("Failed to test query",
+                status,
+                unwrapped));
+        }));
     }
 
     public void deduceMappings(Client client, SourceConfig sourceConfig, final ActionListener<Map<String, String>> listener) {

